contracts/contracts/lib/AssetHelper.sol

```diff
@@ -60,35 +60,40 @@ library SupplySourceHelper {
             }
             // Safe arithmetic is not necessary because underflow is not possible due to the check above
             return finalBalance - initialBalance;
+        } else {
+            // now we are handling native token
+            if (msg.value < value) {
+                revert NoBalanceIncrease();
+            }
```

Comment:
if `msg.value == value`, we should pass the check. If `msg.value > value`, someone wants to do charity, maybe can just take it...

Reply:
Best not to. It's stronger to have an invariant that gateway balance across all tokens and the native coin is zero for initialized subnets, than to have one where it can be non zero.

Reply:
Oh, I realized we have to use `msg.value > value`. The reason is caller might have msg.value more than just locked value, for other purposes as well. In that way, we can only enforce msg.value > value.